app/code/Magento/Sales/Model/Order/Invoice.php

```diff
@@ -270,7 +270,20 @@ public function getBillingAddress()
      */
     public function getShippingAddress()
     {
-        return $this->getOrder()->getShippingAddress();
+        return $this->_findAddress($this->getShippingAddressId()) ?: $this->getOrder()->getShippingAddress();
+    }
+
+    /**
+     * Fetch order address by id
+     *
+     * @return Address|null
+     */
+    protected function _findAddress($addressId) {
+        foreach($this->getOrder()->getAddresses() as $address) {
```

Comment:
$order->getShippingAddress() also uses getAddressses() internally then does an in memory fiter, so this should not change the performance characteristics of this method.